cypress/integration/device_page_test.js

```diff
@@ -0,0 +1,36 @@
+describe('Device page tests', () => {
+  context('no access before login', () => {
+    it('redirects to home', () => {
+      cy.visit('/devices/PO6RuqIXQuysOCAJrcedqA');
+      cy.location('pathname').should('eq', '/login');
+    });
+  });
+
+  context('authenticated', () => {
+    beforeEach(() => {
+      cy.login();
+      cy.fixture('device').as('device');
+      cy.server();
+      cy.route('GET', '/appengine/v1/*/devices/*', '@device');
+      cy.route('GET', '/sockjs-node/**').as('websocket');
+    });
+
+    it('successfully loads', () => {
+      cy.visit('/devices/PO6RuqIXQuysOCAJrcedqA', {
+        onBeforeLoad: (window) => {
+          window.addEventListener('unhandledrejection', (event) => {
+            throw new Error(event.reason);
```

Comment:
tests shouldn't change the page behaviour, if you want to check for connections, check that the message "joined room" in the bottom event log

Reply:
Mocking websocket connection is not straightforward / well-supported in Cypress. I would leave those checks for another PR